### DISCUSSION TITLE
Is v2 compatible with react v16?

### DISCUSSION THREAD

**Discussion by jeannemtl** [WalletConnect/walletconnect-monorepo#2812] [Developer Support] [answered]
Is v2 compatible with react v16? 
Are v2 walletconnect clients instantiated with web3modal only? 
Can I install web2modal v1 to instantiate a client?

**Comment by xzilja** [comment, marked as answer]
You can use `@walletconnect/modal` and `@walletconnect/ethereum-provider` that have no restrictions on react version. Web3Modal is now a more advanced / newer product that supports react 17 and up (mostly due to wagmi requirements)
